flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/DefaultContext.java

```diff
@@ -142,11 +151,13 @@ public static DefaultContext load(
         FileSystem.initialize(
                 configuration, PluginUtils.createPluginManagerFromRootFolder(configuration));
 
-        if (discoverPythonDependency) {
-            dependencies = new ArrayList<>(dependencies);
-            dependencies.addAll(discoverPythonDependencies());
+        if (!discoverExecutionConfig) {
```

Comment:
I felt that the naming `discoverExecutionConfig` is a little hard to understand, why this param can control the python dependency discovery.

Reply:
Ok. I add another parameter to manage the behavior separately.